core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,141 @@
+    String principal = "";
+    String keytab = "";
+    String primary;
+    String instance;
+    String realm;
+    if (mProcess.equals("master")) {
+      principal = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.MASTER_PRINCIPAL.getName()}).getOutput();
+      keytab = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.MASTER_KEYTAB_KEY_FILE.getName()}).getOutput();
+    } else if (mProcess.equals("worker")) {
+      principal = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.WORKER_PRINCIPAL.getName()}).getOutput();
+      keytab = getResultFromProcess(new String[] {ALLUXIO_BIN, GETCONF_CMD,
+          PropertyKey.WORKER_KEYTAB_FILE.getName()}).getOutput();
+    }
+    Matcher matchPrincipal = PRINCIPAL_PATTERN.matcher(principal);
+    if (!matchPrincipal.matches()) {
+      System.err.format("Principal %s is not in the right format.%n", principal);
+      return false;
+    }
+    primary = matchPrincipal.group("primary");
+    instance = matchPrincipal.group("instance");
+    realm = matchPrincipal.group("realm");
+
+    // Login with principal and keytab
+    int exitVal =
+        getResultFromProcess(new String[] {"kinit", "-kt", keytab, principal}).getExitValue();
+    if (exitVal != 0) {
+      System.err.format("Kerberos login failed for %s with keytab %s with exit value %d.%n",
+          principal, keytab, exitVal);
+      System.err.format("Primary is %s, instance is %s and realm is %s.%n",
+          primary, instance, realm);
+      return false;
+    }
+    return true;
+  }
+
+  private ProcessExecutionResult getResultFromProcess(String[] args) {
+    try {
+      Process process = Runtime.getRuntime().exec(args);
+      StringBuilder outputSb = new StringBuilder();
+      try (BufferedReader processOutputReader = new BufferedReader(
+          new InputStreamReader(process.getInputStream()))) {
+        String line;
+        while ((line = processOutputReader.readLine()) != null) {
+          outputSb.append(line);
+          outputSb.append(LINE_SEPARATOR);
+        }
+      }
+      StringBuilder errorSb = new StringBuilder();
+      try (BufferedReader processErrorReader = new BufferedReader(
+          new InputStreamReader(process.getErrorStream()))) {
+        String line;
+        while ((line = processErrorReader.readLine()) != null) {
+          errorSb.append(line);
+          errorSb.append(LINE_SEPARATOR);
+        }
+      }
+      process.waitFor();
+      return new ProcessExecutionResult(process.exitValue(), outputSb.toString().trim(),
+          errorSb.toString().trim());
+    } catch (IOException e) {
+      System.err.println("Failed to execute command.");
+      return new ProcessExecutionResult(1, "", "");
+    } catch (InterruptedException e) {
+      System.err.println("Interrupted.");
+      Thread.currentThread().interrupt();
+      return new ProcessExecutionResult(1, "", "");
```

Comment:
Would be great to include exception message in both stderr and return value

Reply:
I think exception handling should happen at a higher-level. `ProcessExecutionResult` should contain exactly the output and error from the process. Then the upper-level can